### PR TITLE
 Added function for setting up FEP simulations

### PR DESCRIPTION
**Pull request by PicoCentauri**
The function in the setup module allows to set up free energy pertubation simualtions. It will create a single simulation folder for all lambda states to run them individually.

In contrast to the other setup functions I used a dictionary input for the submission script, since I think its more convenient. 

Furthermore I wondered about two thinks: 

1. The setup functions seem not really consistent regarding their input parameters and their standard values between them: for example [energy_miniztion](https://github.com/Becksteinlab/GromacsWrapper/blob/d96c34f5d7d81175e823746322f2760c31447e81/gromacs/setup.py#L609-L655) and [MD](https://github.com/Becksteinlab/GromacsWrapper/blob/d96c34f5d7d81175e823746322f2760c31447e81/gromacs/setup.py#L1034-L1087). Maybe we should think about to unify those. 

2. Is there a reason that the [edit_mdp](https://github.com/Becksteinlab/GromacsWrapper/blob/d96c34f5d7d81175e823746322f2760c31447e81/gromacs/cbook.py#L778-L899) function is not using the mdp file format parser?

### CODE REVIEW

**Comment by orbeckst** [comment]
> Is there a reason that the edit_mdp function is not using the mdp file format parser?

No, just history... the parser came later and I never got around to rewrite something that seemed to work even it was crappy code.

**Comment by PicoCentauri** [comment]
> No, just history... the parser came later and I never got around to rewrite something that seemed to work even it was crappy code. 

Okay, I can take a look to decrease "duplicated" code and reduce probable sources of errors.

**Comment by orbeckst** [comment]
@PicoCentauri sorry to revive an old PR after such a long time – if you're still interested in getting this function in then the main thing would be to have some sort of tests. That's not that easy but would really be good.

Btw, I would very much encourage using the MDP parser instead of of `edit_mdp()`...

File: gromacs/setup.py [hunk 1]
```diff
@@ -1031,6 +1035,93 @@ def MD_restrained(dirname='MD_POSRES', **kwargs):
...
+            
+            energy_minimize(dirname=dirname_state, mdp=em_mdp,
+                            struct=struct_state, top=top, 
+                            output='em.gro', deffnm="em",
+                            init_lambda_state=l, **kwargs)
+            struct_state = os.path.join(dirname_state,"em.gro")
+        
+        new_kwargs = _setup_MD(dirname=dirname_state, mdp=mdp, struct=struct_state, 
+                               top=top, ndx=ndx, mainselection=mainselection, 
+                               init_lambda_state=l, **qsub_opts, **kwargs)
```
Comment (orbeckst) [inline gromacs/setup.py]: Double `**kw1, **kw2` does not work in Python 2.7 – needs to be rewritten.

**Comment by PicoCentauri** [comment]
@orbeckst Never mind for reviving this PR. In principle yes I'm still willing to merge this PR. I will take a look at the issues and the tests this week.

**Comment by PicoCentauri** [comment]
Sorry currently I have no time to keep this PR a live and add sufficient tests.

**Pull request closed without merging**